app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -608,12 +610,40 @@ public void onBackPressed() {
 
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
+        final int keyCode = event.getKeyCode();
```

Comment:
Can we move the `dispatchKeyEvent` implementation to keyboard widget? VRBrowserActivity is becoming big

Reply:
I didn't want to move all of it, but the part relevant to the physical keyboard was easy to move.